includes/class-wc-payments.php

```diff
@@ -234,6 +235,9 @@ public static function init() {
 		// Priority 5 so we can manipulate the registered gateways before they are shown.
 		add_action( 'woocommerce_admin_field_payment_gateways', [ __CLASS__, 'hide_gateways_on_settings_page' ], 5 );
 
+		require_once __DIR__ . '/migrations/class-allowed-payment-request-button-types-update.php';
+		add_action( 'woocommerce_woocommerce_payments_updated', [ new Allowed_Payment_Request_Button_Types_Update( self::get_gateway() ), 'maybe_migrate' ] );
```

Comment:
(minor) It looks like in other cases it's up to the class' constructor to register the hooks and the actions the class subscribes to.
So in this case, we would do `add_action` in `Allowed_Payment_Request_Button_Types_Update`'s constructor. 

```
// includes/class-wc-payments.php
require_once __DIR__ . '/migrations/class-allowed-payment-request-button-types-update.php';
new Allowed_Payment_Request_Button_Types_Update( self::get_gateway() )

// class-allowed-payment-request-button-types-update.php
add_action( 'woocommerce_woocommerce_payments_updated', [ $this, 'maybe_migrate' ] );
```

Was there a specific reason to register the action here rather than in the class' constructor?

Reply:
Right! I tend to avoid doing side effects in constructors so objects can be instantiated (esp. in tests) without them also hooking into actions or filters or doing other things. I can absolutely change this if you feel like the consistency is worth it.

Reply:
Thanks for the explanation! Let's keep it as is :)